urlfb310a588a_alokhan_LinkStateNetworkSimulation_tgz-pJ8-reso_examples_static_routing_DemoJ8/src/reso/scheduler/Scheduler.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * *****************************************************************************
+ * Copyright (c) 2011 Bruno Quoitin. All rights reserved. This program and the
+ * accompanying materials are made available under the terms of the GNU Lesser
+ * Public License v2.1 which accompanies this distribution, and is available at
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+ *
+ * Contributors: Bruno Quoitin - initial API and implementation
+ *****************************************************************************
+ */
+package reso.scheduler;
+
+import java.util.PriorityQueue;
+
+public class Scheduler
+        extends AbstractScheduler {
+
+    private PriorityQueue<AbstractEvent> pq;
+
+    public Scheduler() {
+        this.pq = new PriorityQueue<AbstractEvent>();
+    }
+
+    public void schedule(AbstractEvent evt) {
+        assert (evt.getTime() > getCurrentTime());
+        pq.offer(evt);
+    }
+
+    public boolean hasMoreEvents() {
+        return (pq.size() > 0);
+    }
+
+    public void runNextEvent() throws Exception{
+        AbstractEvent evt = pq.poll();
+        time = evt.getTime();
+      	try {
+    		evt.run();
+    	} catch (Exception e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)